igraph: 0.7.1 -> 0.8.2

MostAwesomeDude opened the pull request:
Refs #90840.

The check phase is now enabled. All tests pass on my machine.

I realized while doing this bump that igraph vendors OpenBLAS, ARPACK,
and GLPK. We can use our versions instead, and igraph builds
successfully, but 34 tests fail. I'm choosing correctness instead of
build times.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

MostAwesomeDude committed:
igraph: 0.7.1 -> 0.8.2

Refs #90840.

The check phase is now enabled. All tests pass on my machine.

I realized while doing this bump that igraph vendors OpenBLAS, ARPACK,
and GLPK. We can use our versions instead, and igraph builds
successfully, but 34 tests fail. I'm choosing correctness instead of
build times.

MostAwesomeDude commented:
Looks like tests also fail on aarch64. If that's a blocker, then we can disable tests, and in that case, we ought to use nixpkgs instead of vendored libraries. All of that is possible; LMK.

Ma27 commented:
It should be fine to disable the tests on aarch64 if *only* the tests are broken. If the package doesn't work there or isn't supported there, we should restrict `meta.platforms`.

MostAwesomeDude committed:
igraph: Only build and test x86_64-linux.

aarch64-linux tests fail on the community builder, so don't build there
until we have more evidence that failing tests are not a problem.

MostAwesomeDude committed:
python-igraph: Don't use vendored library.

Fixes #70543.
